src/z2m/mod.rs

```diff
@@ -188,6 +188,29 @@ impl Client {
             room_name = &grp.friendly_name;
         }
 
+        let topic = grp.friendly_name.to_string();
+        let mut metadata = RoomMetadata::new(RoomArchetype::Home, room_name, false);
+        if let Some(room_conf) = self.config.rooms.get(&topic) {
+            if let Some(name) = &room_conf.name {
+                metadata.name = name.to_string();
+            }
+            if let Some(icon) = &room_conf.icon {
+                metadata.archetype = *icon;
+            }
+            if let Some(hidden) = &room_conf.hidden {
+                metadata.hidden = *hidden;
+            }
+        };
+
+        if metadata.hidden {
```

Comment:
As mentioned above, this is a no-go. We absolutely cannot leak internal details into the hue api.

But the good news is that we already have access to the `config`, so it should be doable to find the information (same way we find name override and icon override).